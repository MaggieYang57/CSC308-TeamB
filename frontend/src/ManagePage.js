import React from "react";
import AdminReviewTable from "./components/AdminReviewTable";
import "bootstrap/dist/css/bootstrap.min.css";
import "./css/ProfilePage.css";

require('dotenv').config()
const backendHostURL = process.env.REACT_APP_BACKEND_HOST_URL

class ManagePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    fetch(`${backendHostURL}/login/${localStorage.getItem("_id")}`)
      .then((res) => res.json())
      .then((data) => {
        console.log("data", data);
        this.setState({ ...data[0] });
      });
  }

  render() {
    return (
      <div>
        <h1 className="header">
          Hello, Admin!
        </h1>
        <div className="reviews" style={{marginTop: '-50px'}}>
          <h2 className="reviews-title">
            <hr />
            List of All Reviews
            <hr />
          </h2>
          <AdminReviewTable reviewList={this.state.user_email} route={""} />
        </div>
      </div>
    );
  }
}

export default ManagePage;
