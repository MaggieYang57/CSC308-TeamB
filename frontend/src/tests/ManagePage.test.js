import React from 'react';
import { cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ManagePage from '../ManagePage';
import AdminUserReviewTable from '../components/AdminUserReviewTable';



function createFetchSpy(fetchRes, jsonRes) {
    const fetchSpy = jest.spyOn(window, 'fetch')
    const jsonSpy = jest.fn();
    
    fetchSpy.mockReturnValue(new Promise((resolve, reject) => {
      resolve({
        json: jsonSpy,
        ...fetchRes
      });
    }))

    jsonSpy.mockReturnValue(new Promise((resolve, reject) => {
      resolve(jsonRes);
    }))

    return [fetchSpy, jsonSpy];
}

describe("Manage Page", () => {
  let fetchSpy;
  let jsonFn;

  beforeAll(() => {
    jest.mock(AdminUserReviewTable, () => () => <div id="Mock">Mocked</div>)
  })

  beforeEach(() => {
    [fetchSpy, jsonFn] = createFetchSpy({
        ok: true,
      }, [{ test: ['hello'] }]);
  })

  afterEach(() => {
    cleanup();
  });

  it('renders without crashing', async () => {
    render(<ManagePage />)

    await waitFor(() => {
      expect(fetchSpy).toHaveBeenCalledTimes(1);
    })

    expect(jsonFn).toHaveBeenCalledTimes(1);
  })

});