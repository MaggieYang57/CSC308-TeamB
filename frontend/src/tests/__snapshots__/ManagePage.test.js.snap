// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage Page snapshot test 1`] = `
<div>
  <h1
    className="header"
  >
    Hello, Admin!
  </h1>
  <div
    className="reviews"
    style={
      Object {
        "marginTop": "-50px",
      }
    }
  >
    <h2
      className="reviews-title"
    >
      <hr />
      List of All Reviews
      <hr />
    </h2>
    <div />
  </div>
</div>
`;
